src/MSRIOGroup.cpp

```diff
@@ -522,10 +522,19 @@ namespace geopm
 
     void MSRIOGroup::save_control(void)
     {
+        std::vector<std::string> unallowed_controls;
         for (auto &ctl : m_control_available) {
-            for (auto &dom_ctl : ctl.second.controls) {
-                dom_ctl->save();
+            try {
+                for (auto &dom_ctl : ctl.second.controls) {
+                    dom_ctl->save();
+                }
             }
+            catch (const Exception &) {
```

Comment:
Any exception that occurs in the ```dom_ctl->save();``` call path will be stopped here.  For MSR based controls, I don't see how any other exceptions could be generated here, so this is fine.  If we need to use this pattern anywhere else, trapping ```Exception``` might be too heavy handed and we should instead use one of the custom GEOPM types.